### PR TITLE
ISSUE #1793: Connect new federal agency table to domain requests and admin

### LINKED ISSUE #1793: Connect Federal Agency drop down with the federal agency table
### Current Behavior
We recently added the federal agencies table, but it is not connected to anything, we should connect it to in the "Organization name and mailing address" (for org type "Federal") request page and anywhere else the federal agency field is used. 
Note in the current state in the "Organization name and mailing address" (for org type "Federal") request page, the current "Federal agency" dropdown shows duplicate entries which do _not_ exist in the federal agency table (they were removed intentionally). There is also an invalid option in the dropdown: "State, Local, and Tribal Government" which should be just Non-Federal Agency instead.  These duplicate and incorrect values do **not** exist in the Federal Agency table; however, there is a migration script ticket (#1975)  to update any 'deprecated' values to the correct ones to use.  If you pick up this current ticket, the migration ticket is a logical choice to work at the same time. They should be merged around the same time.

There are also 2 values we should keep but not allow end users to select:
- gov Administration
- Non-Federal Agency 

### Acceptance Criteria 
- [ ] Make all occurrences of the federal_agency drop down use the new Agency field that is connected to the federal agency table see #2029 and #1896
- [ ] Hide 2 values when shown in the request form's dropdown (but keep them in the database/Django admin):
  - [ ] gov Administration
  - [ ] Non-Federal Agency
- [x] ~In the request form's Federal Agency dropdown, include an "other" field to allow users to add an agency not in our list (ignore this one)~
- [ ] Make sure federal agency on domain request and domain info are all still drop downs
- [ ] ~remove the old federal_agency field  as it is no longer needed and be sure the now linked Agency field now shows on the django admin exactly where federal_agency was showing and that the field is viewable for all users (note the goal is to make it look like nothing changed for analysts)~
- [ ] Create 2 migration files:  1: Migration file to DELETE `federal_agency` 2: Migration file to RENAME `updated_federal_agency` to `federal_agency` (updated_federal_agency now filled in at this point, so we have to delete + rename)
- [ ] Remove tests testing that updated_federal_agency duplicates the FederalAgency version of federal_agency that reference updated_federal_agency as this field will be renamed to federal_agency to replace the current federal_agency field.
- [ ] Make sure both Superuser and Analysts can view the correct new Federal Agency list without issues

### Additional Context

[Slack thread discussion](https://cisa-corp.slack.com/archives/C05BGB4L5NF/p1708109328529279)

### Issue Links
blocked by #1896 
blocked by #2029 
Relates to #1975 (do at around the same time to avoid issues)
Relates to: #337

### PR DESCRIPTION
**Pull request by erinysong**
## Ticket

Resolves #1793

## Changes

- Transition `federal_agency` field from a character field to a foreign key field (fk field formerly called `updated_federal_agency`) that references our new Federal Agency table. Update tests to reflect `federal_agency` change to foreign key field.
- Update fixtures to populate local seed data reflecting `federal_agency` being a fk referencing FederalAgency table.
- Remove list of string constants used to populate `federal_agency` when it was formerly a character field.
- Delete tests for `patch_federal_agency_info` script. This script is expected to have a one-time use and references outdated org model architecture from before we redesigned our org model. The script was used to prepare domain data when passing it into our transition domain scripts that transitioned our domain data from Verisign to Cloudflare. 

## Context for reviewers

Our new org model transition involves creating db tables to store organizations associated with a domain (compared to when we originally stored organization names as plain text fields).  

#1928 created a FederalAgency table to standardize federal agencies associated with a domain. #2046 and #2069 added to `DomainInformation` and `DomainRequest` objects a new field `updated_federal_agency` which assigns domain information and domain requests a FederalAgency object that matches its current text-based federal_agency. The temporary field `updated_federal_agency` was used instead of directly changing `federal_agency` to avoid accidentally losing any production data on stable.

Now that our FederalAgency foreign key is verified to work, this PR completes the FederalAgency portion of the new org model by replacing our text-based federal agency field with a new foreign key field that references the FederalAgency table. 

## Setup

Go to https://getgov-es.app.cloud.gov/ and verify the following.

**Admin: Federal Agency field successfully links to FederalAgency table**
1. Go to /admin and visit and DomainInformation or DomainRequest record. Scroll down to Type of Organization and click `Show details` collapse. This should display the updated federal_agency field.
image
2. Select a federal agency for the domain if there is not one assigned already. Click on the View selected federal agency eye icon.
image
3. You should be routed to the FederalAgency table record of the DomainInformation or DomainRequest's federal agency.
image

**Domain request form: Federal Agency field successfully links to FederalAgency table**

1. On a domain request form, navigate to the Organization name and mailing address page. The Federal Agency dropdown should now be updated so that its outdated values are removed. The list of deprecated agency names can be found below.
image

| Keep  | Remove |
| ------------- | ------------- |
| Appraisal Subcommittee of the Federal Financial Institutions Examination Council  | Appraisal Subcommittee |
| Barry Goldwater Scholarship and Excellence in Education Foundation | Barry Goldwater Scholarship and Excellence in Education Program |
| Federal Reserve Board of Governors | Federal Reserve System |
| Harry S. Truman Scholarship Foundation | Harry S Truman Scholarship Foundation |
| Japan-U.S. Friendship Commission _(create new, replace existing)_ | Japan-US Friendship Commission _and_ Japan-United States Friendship Commission _(remove both)_ | 
| John F. Kennedy Center for the Performing Arts | John F. Kennedy Center for Performing Arts |
| Occupational Safety and Health Review Commission | Occupational Safety & Health Review Commission |
| Corporation for National and Community Service | Corporation for National & Community Service |
| Export-Import Bank of the United States | Export/Import Bank of the U.S. |
| Medicare Payment Advisory Commission | Medical Payment Advisory Commission |
| Peace Corps | U.S. Peace Corps |
| U.S. Chemical Safety Board | Chemical Safety Board |
| U.S. Nuclear Waste Technical Review Board | Nuclear Waste Technical Review Board |
| State, Local, & Tribal Government | _fully removed_ |

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [x] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/fixtures_domain_requests.py [hunk 1]
```diff
@@ -146,6 +134,12 @@ def _set_foreign_key_fields(cls, da: DomainRequest, app: dict, user: User):
                 da.requested_domain, _ = DraftDomain.objects.get_or_create(name=app["requested_domain"])
             else:
                 da.requested_domain = DraftDomain.objects.create(name=cls.fake_dot_gov())
+        if not da.federal_agency:
+            if "federal_agency" in app and app["federal_agency"] is not None:
+                da.federal_agency, _ = FederalAgency.objects.get_or_create(name=app["federal_agency"])
+            else:
+                federal_agencies = FederalAgency.objects.all()
+                da.federal_agency = random.choice(federal_agencies)
```
Comment (abroddrick) [inline src/registrar/fixtures_domain_requests.py:142]: Thanks for fixing this in fixtures too!

File: src/registrar/fixtures_domain_requests.py [hunk 2]
```diff
@@ -146,6 +134,13 @@ def _set_foreign_key_fields(cls, da: DomainRequest, app: dict, user: User):
                 da.requested_domain, _ = DraftDomain.objects.get_or_create(name=app["requested_domain"])
             else:
                 da.requested_domain = DraftDomain.objects.create(name=cls.fake_dot_gov())
+        if not da.federal_agency:
+            if "federal_agency" in app and app["federal_agency"] is not None:
+                da.federal_agency, _ = FederalAgency.objects.get_or_create(name=app["federal_agency"])
+            else:
+                federal_agencies = FederalAgency.objects.all()
+                # Random choice of agency for selects, used as placeholders for testing.
+                da.federal_agency = random.choice(federal_agencies)  # nosec
```
Comment (erinysong) [inline src/registrar/fixtures_domain_requests.py:143]: # nosec is used in all references to random.choice to seed local fixtures.

File: src/registrar/fixtures_domain_requests.py [hunk 3]
```diff
@@ -3,13 +3,7 @@
...
 from django.db import transaction
 
-from registrar.models import (
-    User,
-    DomainRequest,
-    DraftDomain,
-    Contact,
-    Website,
-)
+from registrar.models import User, DomainRequest, DraftDomain, Contact, Website, FederalAgency
```
Comment (CocoByte) [inline src/registrar/fixtures_domain_requests.py:6]: (nonblocking) Was this a linter thing?  Kindof favored the more readable layout.  Personal opinion, really.  Some coders prefer compact.

**Review by CocoByte — APPROVED** [review]
LGTM!  Overall, really well-written PR.  Made reviewing super smooth.  I like the refactor work that stripped out all those hard-coded strings.

Tested and verified on sandbox.

File: src/registrar/fixtures_domain_requests.py (hunk 3 above)
Comment (therealslimhsiehdy) [reply src/registrar/fixtures_domain_requests.py:6]: I don't recall us editing this, so I believe it was a linter thing!

**Pull request merged**